src/config.js

```diff
@@ -41,15 +41,33 @@ export const flags = {
     default: true,
     icon: '🐳',
   },
-};
-
-export const sidebar = {
-  Categories: {
-    Tags: 'tag',
-  },
-  Elements: {
-    Nodes: 'task',
-    Datasets: 'data',
-    Parameters: 'parameters',
+  modularpipeline: {
+    description: 'enables modular pipeline features',
+    default: false,
+    icon: '⛓️',
   },
 };
+
+export const sidebar = (flag) =>
```

Comment:
Thanks for the suggestion Richard! I definitely had intended and would've set it up in your suggested way but unfortunately it would not work with the existing sidebar setup ( having a field as `undefined` would cause errors in the sidebar). 

If course we could refactor the sidebar to change how we accept the sidebar config object, but given that hiding this behind a flag is a temporary measure ( and will subject to iteration in the next round) I had decided to keep it as is.

Reply:
Yeah a static config is definitely preferred generally, so would be good to add support for this case at the source if practical. 

Flags do make this tricky though so if that's not straightforward then maybe this pattern gets closer to the above suggestion e.g.

```js
export const sidebar = (flags) => ({
  Categories: {
    Tags: 'tag',
    ...(flags.modularPipeline && { 
      ModularPipelines: 'modularPipeline' 
    }),
  },
  Elements: {
    Nodes: 'task',
    Datasets: 'data',
    Parameters: 'parameters',
  },
})
```